catlin/pkg/validator/task_validator.go

```diff
@@ -0,0 +1,51 @@
+// Copyright © 2020 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validator
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"go.uber.org/zap"
+
+	"github.com/tektoncd/plumbing/catlin/pkg/parser"
+)
+
+type TaskValidator struct {
+	log *zap.SugaredLogger
+	res *parser.Resource
+}
+
+var _ Validator = (*TaskValidator)(nil)
+
+func (t *TaskValidator) Validate() Result {
+
+	result := Result{}
+
+	var task *v1beta1.Task
+	str, _ := json.Marshal(t.res.Unstructured.Object)
```

Comment:
I think it may be better to add a function to the `ParsedResource` to convert to the right type. 
```go
var task v1beta.Task
err := r.ToType(&task)
```
and then 

```
for _, step := ... 
```

This makes it quite easy to add  support for other types as well. Please refer to the code in parse to properly unmarshall to a typed object.

Reply:
Thank you :)